{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pathlib\n",
    "\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['DATE ADDED TO CATALOG', 'PUBMEDID', 'FIRST AUTHOR', 'DATE', 'JOURNAL',\n",
      "       'LINK', 'STUDY', 'DISEASE/TRAIT', 'INITIAL SAMPLE SIZE',\n",
      "       'REPLICATION SAMPLE SIZE', 'REGION', 'CHR_ID', 'CHR_POS',\n",
      "       'REPORTED GENE(S)', 'MAPPED_GENE', 'UPSTREAM_GENE_ID',\n",
      "       'DOWNSTREAM_GENE_ID', 'SNP_GENE_IDS', 'UPSTREAM_GENE_DISTANCE',\n",
      "       'DOWNSTREAM_GENE_DISTANCE', 'STRONGEST SNP-RISK ALLELE', 'SNPS',\n",
      "       'MERGED', 'SNP_ID_CURRENT', 'CONTEXT', 'INTERGENIC',\n",
      "       'RISK ALLELE FREQUENCY', 'P-VALUE', 'PVALUE_MLOG', 'P-VALUE (TEXT)',\n",
      "       'OR or BETA', '95% CI (TEXT)', 'PLATFORM [SNPS PASSING QC]', 'CNV',\n",
      "       'MAPPED_TRAIT', 'MAPPED_TRAIT_URI', 'STUDY ACCESSION',\n",
      "       'GENOTYPING TECHNOLOGY'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DATE ADDED TO CATALOG</th>\n",
       "      <th>PUBMEDID</th>\n",
       "      <th>FIRST AUTHOR</th>\n",
       "      <th>DATE</th>\n",
       "      <th>JOURNAL</th>\n",
       "      <th>LINK</th>\n",
       "      <th>STUDY</th>\n",
       "      <th>DISEASE/TRAIT</th>\n",
       "      <th>INITIAL SAMPLE SIZE</th>\n",
       "      <th>REPLICATION SAMPLE SIZE</th>\n",
       "      <th>...</th>\n",
       "      <th>PVALUE_MLOG</th>\n",
       "      <th>P-VALUE (TEXT)</th>\n",
       "      <th>OR or BETA</th>\n",
       "      <th>95% CI (TEXT)</th>\n",
       "      <th>PLATFORM [SNPS PASSING QC]</th>\n",
       "      <th>CNV</th>\n",
       "      <th>MAPPED_TRAIT</th>\n",
       "      <th>MAPPED_TRAIT_URI</th>\n",
       "      <th>STUDY ACCESSION</th>\n",
       "      <th>GENOTYPING TECHNOLOGY</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>2019-07-12</td>\n",
       "      <td>31049640</td>\n",
       "      <td>Chen J</td>\n",
       "      <td>2019-05-02</td>\n",
       "      <td>Diabetologia</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/31049640</td>\n",
       "      <td>Genome-wide association study of type 2 diabet...</td>\n",
       "      <td>Type 2 diabetes</td>\n",
       "      <td>2,633 African ancestry cases, 1,714 African an...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>5.397940</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.34</td>\n",
       "      <td>[1.17-1.53]</td>\n",
       "      <td>Affymetrix, Illumina [12148595] (imputed)</td>\n",
       "      <td>N</td>\n",
       "      <td>type II diabetes mellitus</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0001360</td>\n",
       "      <td>GCST008114</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>2019-07-12</td>\n",
       "      <td>31049640</td>\n",
       "      <td>Chen J</td>\n",
       "      <td>2019-05-02</td>\n",
       "      <td>Diabetologia</td>\n",
       "      <td>www.ncbi.nlm.nih.gov/pubmed/31049640</td>\n",
       "      <td>Genome-wide association study of type 2 diabet...</td>\n",
       "      <td>Type 2 diabetes</td>\n",
       "      <td>2,633 African ancestry cases, 1,714 African an...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>5.045757</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.84</td>\n",
       "      <td>[1.42-2.38]</td>\n",
       "      <td>Affymetrix, Illumina [12148595] (imputed)</td>\n",
       "      <td>N</td>\n",
       "      <td>type II diabetes mellitus</td>\n",
       "      <td>http://www.ebi.ac.uk/efo/EFO_0001360</td>\n",
       "      <td>GCST008114</td>\n",
       "      <td>Genome-wide genotyping array</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows Ã— 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  DATE ADDED TO CATALOG  PUBMEDID FIRST AUTHOR        DATE       JOURNAL  \\\n",
       "0            2019-07-12  31049640       Chen J  2019-05-02  Diabetologia   \n",
       "1            2019-07-12  31049640       Chen J  2019-05-02  Diabetologia   \n",
       "\n",
       "                                   LINK  \\\n",
       "0  www.ncbi.nlm.nih.gov/pubmed/31049640   \n",
       "1  www.ncbi.nlm.nih.gov/pubmed/31049640   \n",
       "\n",
       "                                               STUDY    DISEASE/TRAIT  \\\n",
       "0  Genome-wide association study of type 2 diabet...  Type 2 diabetes   \n",
       "1  Genome-wide association study of type 2 diabet...  Type 2 diabetes   \n",
       "\n",
       "                                 INITIAL SAMPLE SIZE REPLICATION SAMPLE SIZE  \\\n",
       "0  2,633 African ancestry cases, 1,714 African an...                     NaN   \n",
       "1  2,633 African ancestry cases, 1,714 African an...                     NaN   \n",
       "\n",
       "   ... PVALUE_MLOG P-VALUE (TEXT)  OR or BETA 95% CI (TEXT)  \\\n",
       "0  ...    5.397940            NaN        1.34   [1.17-1.53]   \n",
       "1  ...    5.045757            NaN        1.84   [1.42-2.38]   \n",
       "\n",
       "                  PLATFORM [SNPS PASSING QC] CNV               MAPPED_TRAIT  \\\n",
       "0  Affymetrix, Illumina [12148595] (imputed)   N  type II diabetes mellitus   \n",
       "1  Affymetrix, Illumina [12148595] (imputed)   N  type II diabetes mellitus   \n",
       "\n",
       "                       MAPPED_TRAIT_URI  STUDY ACCESSION  \\\n",
       "0  http://www.ebi.ac.uk/efo/EFO_0001360       GCST008114   \n",
       "1  http://www.ebi.ac.uk/efo/EFO_0001360       GCST008114   \n",
       "\n",
       "          GENOTYPING TECHNOLOGY  \n",
       "0  Genome-wide genotyping array  \n",
       "1  Genome-wide genotyping array  \n",
       "\n",
       "[2 rows x 38 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gwas_files = list(pathlib.Path('../data/gwas_catalog/').glob('*.tsv'))\n",
    "\n",
    "gwas_df = pd.concat(pd.read_csv(path, sep='\\t') for path in gwas_files)\n",
    "\n",
    "gwas_df.to_csv('../data/gwas_catalog/combined.csv', index=False)\n",
    "\n",
    "print(gwas_df.columns)\n",
    "\n",
    "gwas_df.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>STUDY ACCESSION</th>\n",
       "      <th>n_row</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>GCST005047</td>\n",
       "      <td>110</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>GCST007847</td>\n",
       "      <td>126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>GCST008114</td>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  STUDY ACCESSION  n_row\n",
       "0      GCST005047    110\n",
       "1      GCST007847    126\n",
       "2      GCST008114     40"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "(\n",
    "    gwas_df\n",
    "    .groupby('STUDY ACCESSION')\n",
    "    .size()\n",
    "    .reset_index()\n",
    "    .rename(columns={0: 'n_row'})\n",
    ")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:nsides] *",
   "language": "python",
   "name": "conda-env-nsides-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
